### PR TITLE
Remap data docs

### PR DESCRIPTION
**Pull request by megcrenshaw**
This PR remaps data-library urls to use data_doc pages.

Video of the remapping:
https://user-images.githubusercontent.com/9142121/196204855-a58f1bca-01dc-46fe-a7a3-22af04b3e1ca.mov

## Follow-up work

[Look to see where `docs/concepts/data-library` are being used in our curriculum and change their usages to `/data_docs`
](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64/backlog?view=detail&selectedIssue=ACQ-175&issueLimit=100)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/routes.rb [hunk 1]
```diff
@@ -94,6 +95,9 @@
       end
     end
 
+    get 'docs/concepts/data-library', to: 'data_docs#index'
+    get 'docs/concepts/data-library/:data_docs_key', constraints: {data_docs_key: /#{CurriculumHelper::KEY_CHAR_RE}+/o}, to: 'data_docs#show'
+
     get 'docs/*path', to: 'curriculum_proxy#get_doc'
     get 'curriculum/*path', to: 'curriculum_proxy#get_curriculum'
```
Comment (megcrenshaw) [inline dashboard/config/routes.rb:100]: Will we still have the `curriculum_proxy`? Should reference guides be re-routed too?

Comment (dmcavoy) [reply dashboard/config/routes.rb:100]: We will still need the `curriculum_proxy` for the older versions of courses which still point at CB for lesson plans

Comment (dmcavoy) [reply dashboard/config/routes.rb:100]: For reference guides it would probably be good to re-route

Comment (megcrenshaw) [reply dashboard/config/routes.rb:100]: Sounds good. So I think the `docs/*path` doc could be removed, as well as the `get_doc` method

Comment (megcrenshaw) [reply dashboard/config/routes.rb:100]: Update: Since the reference guides are now versioned, we are not going to re-map these.

Comment (bethanyaconnor) [reply dashboard/config/routes.rb:100]: There are some old links in old scripts that point to things like [curriculum.code.org/docs/csd17-18/headers](https://curriculum.code.org/docs/csd-1718/headers/) (for example, https://studio.code.org/s/csd2-2018/lessons/4/levels/6). It sounds like y'all decided to keep this route here anyway, but that's the kind of link that I was thinking of when I kept this catch-all path

Comment (megcrenshaw) [reply dashboard/config/routes.rb:100]: Great, thanks! I'll clarify in a comment below.

Comment (megcrenshaw) [reply dashboard/config/routes.rb:100]: We decided not to try to logic a way for those to point to something like https://studio.code.org/courses/csd-2022/guides/headings-and-paragraphs, so we'll leave the proxy to handle those

File: dashboard/config/routes.rb [hunk 2]
```diff
@@ -94,6 +89,13 @@
       end
     end
 
+    # Data docs are off of curriculum builder as of fall 2022.
+    get 'docs/concepts/data-library', to: 'data_docs#index'
+    get 'docs/concepts/data-library/:key', param: :key, constraints: {data_doc_key: /#{CurriculumHelper::KEY_CHAR_RE}+/o}, to: 'data_docs#show'
+
+    # There are still old curricula that live on curriculum.code.org.
+    # There are also old levels that point to docs on curriculum.code.org.
```
Comment (davidsbailey) [inline dashboard/config/routes.rb:97]: nice updates to comments! 👍 

interestingly, it looks like 90+% of the remaining references to CB docs are pointing to specific IDEs, which are now served by rails via the above routes, and there are just a few pointing to `/docs/csd/...` which look like reference guides that we have not updated yet:
Screen Shot 2022-10-18 at 10 41 50 AM

I think you're good to ship this how it is, but also we are very close to being able to eliminate the `get 'docs/*path'` rule in the future.

**Review by bethanyaconnor — APPROVED** [review]

**Review by davidsbailey — APPROVED** [review]
🚀 ❗

File: dashboard/config/routes.rb (hunk 2 above)
Comment (megcrenshaw) [reply dashboard/config/routes.rb:97]: O, good call! I'll add this to the backlog.

**Pull request merged**